boards/xtensa/intel_adsp_cavs15/tools/adsplog.py

```diff
@@ -40,31 +40,39 @@
 WIN_SIZE = 0x20000
 LOG_OFFSET = WIN_OFFSET + WIN_IDX * WIN_SIZE
 
-# List of known ADSP devices by their PCI IDs
-DEVICES = ["8086:5a98"]
-
 mem = None
+pcid="/sys/bus/pci/devices"
+
+for fname in os.listdir(pcid):
+    pfile = pcid + "/" + fname + "/class"
+    if not os.path.exists(pfile):
+        continue
+    fd = open(pfile)
+    pciclass = fd.read()
```

Comment:
I have extremely little idea about python, but I heard they really root for using "with" which would go something like
```
with fd = open(pfile)
    pciclass = fd.read()
```
and is supposed to handle a tonne of errors and what not. But maybe this isn't applicable here

Reply:
@lyakh that would probably nicer, but I don't see a real functional here, so I'll keep this without "with" for now.